sdk/src/main/scala/com/horizen/account/state/WithdrawalMsgProcessor.scala

```diff
@@ -0,0 +1,195 @@
+    //TODO: check errors in Ethereum, maybe for some kind of errors there a predefined types or codes
+
+    try {
+      if (!canProcess(msg, view)) {
+        new InvalidMessage(new IllegalArgumentException(s"Cannot process message $msg"))
+      }
+      else {
+
+        val functionSig = BytesUtils.toHexString(getOpCodeFromData(msg.getData))
+
+        if (getListOfWithdrawalReqsCmdSig.equals(functionSig)) {
+          getListOfWithdrawalReqRecords(msg, view) match {
+            case Success(result) =>
+              result
+            //TODO log
+            case Failure(exception: IllegalArgumentException) =>
+              new ExecutionFailed(gasSpentForGetListOfWithdrawalReqsFailure, exception)
+            case Failure(exception) =>
+              log.error(s"Error while requesting Withdrawal Request list: ${exception.getMessage}", exception)
+              new ExecutionFailed(gasSpentForGetListOfWithdrawalReqsFailure, new Exception(exception))
+          }
+        }
+        else if (addNewWithdrawalReqCmdSig.equals(functionSig)) {
+          addWithdrawalRequest(msg, view) match {
+            case Success(result) =>
+              result
+            //TODO log
+            case Failure(exception: IllegalArgumentException) =>
+              new ExecutionFailed(gasSpentForAddNewWithdrawalReqFailure, exception)
+            case Failure(exception) =>
+              log.error(s"Error while adding a new Withdrawal Request: ${exception.getMessage}", exception)
+              new ExecutionFailed(gasSpentForAddNewWithdrawalReqFailure, new Exception(exception))
+          }
+        }
+        else {
+          log.error(s"Requested function does not exist. Function signature: $functionSig")
+          new ExecutionFailed(gasSpentForGenericFailure, new IllegalArgumentException(s"Requested function does not exist"))
+        }
+      }
+    }
+    catch {
+      case e: Exception =>
+        log.error(s"Exception while processing message: $msg", e)
+        new ExecutionFailed(gasSpentForGenericFailure, e)
+    }
+
+  }
+
+  private def getWithdrawalEpochCounter(view: AccountStateView, epochNum: Int) = {
+    val key = getWithdrawalEpochCounterKey(epochNum)
+    val wrCounterInBytes = view.getAccountStorageBytes(fakeSmartContractAddress.address(), key).get
+    val numOfWithdrawalReqs = if (wrCounterInBytes.length < Ints.BYTES) {
+      0
+    }
+    else {
+      Ints.fromByteArray(wrCounterInBytes)
+    }
+    numOfWithdrawalReqs
+  }
+
+  override private[horizen] def getListOfWithdrawalReqRecords(epochNum: Int, view: AccountStateView): Seq[WithdrawalRequest] = {
+    val numOfWithdrawalReqs: Int = getWithdrawalEpochCounter(view, epochNum)
+
+    val listOfWithdrawalReqs = (1 to numOfWithdrawalReqs).map(index => {
+      val currentKey = getWithdrawalRequestsKey(epochNum, index)
+      WithdrawalRequestSerializer.parseBytes(view.getAccountStorageBytes(fakeSmartContractAddress.address(), currentKey).get)
```

Comment:
According to the comment above for exceptions handling, why not to catch and log parsing error here?